### PR TITLE
feat: Added redis plugin

### PR DESCRIPTION
**Pull request by frederikpyt**
## Description
1. Added fastify redis plugin, so we can cache stuff on our redis server
2. Added remember function for better usage of redis

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: src/plugins/redis.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    }
+}
+
+export default fastifyPlugin(
+    async (fastify: FastifyInstance, opts: FastifyPluginOptions) => {
+        await fastify.register(fastifyRedis, {
+            host: fastify.config.REDIS_HOST,
+            password: fastify.config.REDIS_PASSWORD,
+            family: 4,
+        });
```
Comment (kristianbinau) [inline src/plugins/redis.ts:17]: Add remember function: `redis.remember(String key, Callback function, Interger ttl): String`

**Review by kristianbinau — CHANGES_REQUESTED** [review]
Only small comment

File: src/plugins/redis.ts (hunk 1 above)
Comment (frederikpyt) [reply src/plugins/redis.ts:27]: Done

File: src/plugins/redis.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        ): Promise<string>;
+    }
+}
+
+export default fastifyPlugin(
+    async (fastify: FastifyInstance, opts: FastifyPluginOptions) => {
+        await fastify.register(fastifyRedis, {
+            host: fastify.config.REDIS_HOST,
+            password: fastify.config.REDIS_PASSWORD,
+            family: 4,
```
Comment (kristianbinau) [inline src/plugins/redis.ts:26]: delete, default value.

File: src/plugins/redis.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        });
+
+        fastify.redis.remember = async (key, ttl, callback) => {
+            let value = await fastify.redis.get(key);
+
+            if (value) {
```
Comment (kristianbinau) [inline src/plugins/redis.ts:32]:
```suggestion
            if (value !== null) {
```

**Review by kristianbinau — CHANGES_REQUESTED** [review]
Small fixes

File: src/plugins/redis.ts (hunk 2 above)
Comment (frederikpyt) [reply src/plugins/redis.ts:26]: OK

**Review by kristianbinau — APPROVED** [review]
It works! 🦺

**Pull request merged**